__tests__/lib/createGHA.test.ts

```diff
@@ -0,0 +1,218 @@
+import fetch from 'node-fetch';
+import prompts from 'prompts';
+
+import OpenAPICommand from '../../src/cmds/openapi';
+import ValidateCommand from '../../src/cmds/validate';
+import createGHA, { cleanUpFileName, git, getGitData } from '../../src/lib/createGHA';
+import getGitRemoteMock from '../helpers/get-git-mock';
+import ghaWorkflowSchemaBackup from '../helpers/github-workflow-schema.json';
+
+const ghaWorkflowUrl = 'https://json.schemastore.org/github-workflow.json';
+
+let consoleInfoSpy;
+const getCommandOutput = () => consoleInfoSpy.mock.calls.join('\n\n');
+
+describe('#createGHA', () => {
+  let ghaWorkflowSchema;
+
+  beforeAll(async () => {
+    ghaWorkflowSchema = await fetch(ghaWorkflowUrl)
+      .then(res => res.json())
+      .catch(() => {
+        console.error('error fetching JSON schema');
+        return ghaWorkflowSchemaBackup;
+      });
+  });
+
+  beforeEach(() => {
+    consoleInfoSpy = jest.spyOn(console, 'info').mockImplementation();
+
+    // global Date override to handle timestamp generation
+    // stolen from here: https://github.com/facebook/jest/issues/2234#issuecomment-294873406
+    const DATE_TO_USE = new Date('2022');
+    // @ts-expect-error we're just overriding the constructor for tests,
+    // no need to construct everything
+    global.Date = jest.fn(() => DATE_TO_USE);
+
+    git.checkIsRepo = jest.fn(() => {
+      return Promise.resolve(true) as unknown as Response<boolean>;
+    });
+
+    git.remote = getGitRemoteMock();
+
+    process.env.TEST_CREATEGHA = 'true';
+  });
+
+  afterEach(() => {
+    consoleInfoSpy.mockRestore();
+    delete process.env.TEST_CREATEGHA;
+    jest.clearAllMocks();
+  });
+
+  describe('command inputs', () => {
+    describe.each([
+      ['validate' as keyof typeof commands, ValidateCommand, { spec: 'petstore.json' } as CommandOptions<{}>],
+      [
+        'openapi' as keyof typeof commands,
+        OpenAPICommand,
+        { spec: 'petstore.json', id: 'spec_id' } as CommandOptions<{}>,
+      ],
+    ])('%s', (cmd, CmdClass, opts) => {
+      let command;
+
+      beforeEach(() => {
+        command = new CmdClass();
+      });
+
+      it('should run GHA creation workflow and generate valid workflow file', async () => {
+        expect.assertions(6);
+        const fileName = `rdme-${cmd}`;
+        prompts.inject([true, 'some-branch', fileName]);
+
+        let yamlOutput;
+
+        fs.writeFileSync = jest.fn((f, d) => {
+          yamlOutput = d;
+          return true;
+        });
+
+        await expect(createGHA('', cmd, command.args, opts)).resolves.toMatchSnapshot();
+
+        expect(yamlOutput).toBeValidSchema(ghaWorkflowSchema);
+        expect(yamlOutput).toMatchSnapshot();
+        expect(fs.writeFileSync).toHaveBeenCalledWith(`.github/workflows/${fileName}.yml`, expect.any(String));
+        expect(console.info).toHaveBeenCalledTimes(1);
+        const output = getCommandOutput();
+        expect(output).toMatch('GitHub Repository detected!');
```

Comment:
For every command that this can be run with, we're running a **ton** of tests:
- YAML file snapshot tests
- Command output snapshot tests
- `fs.writeFileSync` mock tests to ensure it's being called properly
- JSON Schema validation of the YAML output to ensure it conforms to the GitHub Action Workflow specification (see `__tests__/helpers/jest.matchers.ts`)